include/FunctionalFunctorsContainerToContainer.h

```diff
@@ -0,0 +1,837 @@
+                else if constexpr(HasInsertValue<ReturnContainerType>())
+                    { Results.insert(*Copying); }
+                else if constexpr(HasAddValue<ReturnContainerType>())
+                    { Results.add(*Copying); }
+
+                Copying++;
+            }
+
+            return Results;
+        }
+    END_FUNCTOR_END_CONTAINER_TO_CONTAINER_AUTOCURRY(DropWhileStruct, DropWhile)
+
+    /// @struct DropBackWhileStruct
+    /// @details This is an implementation detail for @ref DropBackWhile
+
+    /// @fn DropBackWhile
+    /// @details todo
+    /// @return todo
+
+    START_FUNCTOR_END_CONTAINER_TO_CONTAINER_AUTOCURRY(DropBackWhileStruct)
+        template<typename CVIncomingContainerType, typename PredicateType>
+        [[nodiscard]] constexpr
+        decltype(auto) operator() (const CVIncomingContainerType& IncomingContainer, PredicateType Predicate) const
+        {
+            using IncomingContainerType = std::decay_t<CVIncomingContainerType>;
+            using ActualCVReturnType = std::conditional_t<
+                std::is_same_v<CVReturnContainerType, std::false_type>, IncomingContainerType, CVReturnContainerType
+            >;
+            using ReturnContainerType = std::decay_t<ActualCVReturnType>;
+            using namespace ContainerDetect;
+
+            static_assert(IsRange<IncomingContainerType>(),
+                    "Mezzanine::DropBackWhile only drops from containers or ranges.");
+            static_assert(IsContainer<ReturnContainerType>(),
+                    "Mezzanine::DropBackWhile only drops into containers.");
+            static_assert(HasPushBackValue<ReturnContainerType>() ||
+                          HasInsertValue<ReturnContainerType>() ||
+                          HasAddValue<ReturnContainerType>(),
+                    "Mezzanine::DropBackWhile only drops into containers with push_back(value_type), add(value_type) "
+                    "or insert(value_type).");
+            static_assert(std::is_default_constructible<ReturnContainerType>(),
+                    "Mezzanine::DropBackWhile only drops into containers that can be default constructed.");
+
+            ReturnContainerType Results;
+
+            if constexpr( HasReserve<ReturnContainerType>())
+                { Results.reserve(IncomingContainer.size()); }
+
+            auto Copying = IncomingContainer.crbegin();
+            while(IncomingContainer.crend() != Copying && std::invoke(Predicate, *Copying))
+                { Copying++; }
+
+            while(IncomingContainer.crend() != Copying)
+            {
+                if constexpr(HasPushBackValue<ReturnContainerType>())
+                    { Results.push_back(*Copying); }
+                else if constexpr(HasInsertValue<ReturnContainerType>())
+                    { Results.insert(*Copying); }
+                else if constexpr(HasAddValue<ReturnContainerType>())
+                    { Results.add(*Copying); }
```

Comment:
For all of the operations that copy the elements, should we static_assert to verify the element is copy constructable?  I also have the slightest bit of concern about situations where a moving variant of these functions may be selected instead of the copy.

Reply:
These are const iterators pointing into a const container, moving shouldn't be possible.